### ISSUE TITLE
gpu_normalize() が毎バッチで mean/std テンソルを再生成している

### ISSUE THREAD

**Issue by kurorosu** [kurorosu/pochitrain#197]
## 概要

`gpu_normalize()` は呼び出しのたびに `torch.tensor(mean, device="cuda")` で CUDA テンソルを生成している. 推論ループでは全画像に対して毎回呼ばれるため, テンソル生成のオーバーヘッドが蓄積する.

低レイテンシ最適化を目的とした Plan C パイプラインにおいて, この不要なアロケーションは改善余地がある.

## 問題の詳細

```python
# pochitrain/pochi_dataset.py:293-294
# 毎回 CUDA テンソルを新規生成
mean_255 = torch.tensor(mean, device="cuda").view(1, 3, 1, 1) * 255.0
std_255 = torch.tensor(std, device="cuda").view(1, 3, 1, 1) * 255.0
```

推論ループ内で数百〜数千回呼ばれるため, 毎回のテンソルアロケーション + H2D転送が不要に発生する.

## 修正方針

`mean_255` / `std_255` をループ外で事前計算し, `gpu_normalize()` に引数として渡すか, キャッシュ機構を導入する.

**案A**: 事前計算テンソルを引数で渡す

```python
# CLI側で1回だけ生成
mean_255 = torch.tensor(mean, device="cuda").view(1, 3, 1, 1) * 255.0
std_255 = torch.tensor(std, device="cuda").view(1, 3, 1, 1) * 255.0

# ループ内
gpu_tensor = gpu_normalize(images, mean_255, std_255)
```

**案B**: `functools.lru_cache` やクラス化で内部キャッシュ

## Labels

`enhancement`

## Branch

`refactor/gpu-normalize-cache`
